packages/module/src/HorizontalNav/HorizontalNav.tsx

```diff
@@ -0,0 +1,83 @@
+import { Tabs, Tab, TabTitleText } from '@patternfly/react-core';
+import React from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import '@patternfly/react-styles/css/utilities/Spacing/spacing.css';
+
+export interface TabProps {
+  /** Key for individual tab */
+  key: string | number;
+  /** Title for individual tab */
+  title: string;
+  /** Content for individual tab (provided as a React component) */
+  content: React.ReactElement;
+  /** aria-label for individual tab */
+  ariaLabel: string;
+};
+
+export interface WithRouterProps {
+  /** URL parameters */
+  params?: Record<string, string>;
+  /** Navigate function */
+  navigate?: ReturnType<typeof useNavigate>;
+  /** Current location */
+  location?: ReturnType<typeof useLocation>;
+};
+
+export interface HorizontalNavProps extends WithRouterProps {
+  /** aria-label for all tabs */
+  ariaLabel?: string;
+  /** Properties for tabs */
+  tabs: TabProps[];
+};
+
+const HorizontalNav: React.FunctionComponent<HorizontalNavProps> = ({
+  ariaLabel,
+  tabs,
+  params,
+  navigate,
+  location,
+}: HorizontalNavProps) => {
+  const defaultActiveTab = tabs && tabs[0] ? tabs[0].key : 0; // Set first tab as the default active tab
+
+  const activeTabFromUrlParam = params?.selectedTab;
+  const isValidTabFromUrl =
+    activeTabFromUrlParam && tabs?.some((tab) => tab.key === activeTabFromUrlParam);
+  const activeTab = isValidTabFromUrl ? activeTabFromUrlParam : defaultActiveTab;
+
+  const [ activeTabKey, setActiveTabKey ] = React.useState<string | number>(activeTab);
+
+  return (
+    <Tabs
+      mountOnEnter
+      activeKey={activeTabKey}
+      onSelect={(e, eventKey) => {
+        setActiveTabKey(eventKey);
+        if (location?.pathname && navigate) {
+          const currentPathName = location.pathname;
+          if (params?.selectedTab) {
+            navigate(currentPathName.replace(params.selectedTab, eventKey as string), {
```

Comment:
Hmm. The `Tab` component does support a `href` prop, but we can't customize the root component. 

I think using the `Link` from the router package component is optimal but the `Tab` component support is currently not good enough. If we use it as a child node of the Tab `title` prop, we will require some extra styling. Can you experiment with it to see if we can use it? Let me know if its too much work for too little reward. We can skip it if it is.

Reply:
For starters, I'm unclear how to deal with routing in general in the context of patternfly-docs.  For example, how do I make use of react-router-dom hooks?  Adding them to an example does *not* work.  Perhaps it's not possible?

Reply:
Yeah, that is fair. I see we have two options. We either endorses the react-router-dom library for routing and expect the consuming app to have the router context setup, or we extend the API to not just only use click events to handle navigation but also proved the option to configure root elements of components that support the usage of link tags.

The reason why it's not working in example docs is because the Router context was never set up.

Reply:
@Hyperkid123, I defer to those of you with more experience on the subject.  What do y'all think?  @jhadvig, @florkbr, @fhlavac, @dlabaj

Reply:
We do try to keep this repo as generic as possible. I would personally go with providing the API to modify the components rather than enforcing any libraries. After all, the long-term goal of this repo is to be product agnostic.

That said, there are other files that already use the router hooks etc.

Is it OK if we say, is OK for now, but we will make a breaking change in the near future that would change the component API to allow this generic approach?

Reply:
I agree that the intent of this repo is for reusable UI components that extend PF in a meaningful way. The flows may be specific to Redhat.

@dlabaj can we reuse the PF beta mechanism for releasing these components? That way we can really think through the API changes going forward. I'd imagine this would be valuable to other components we contribute in the future. 

Something like: 
![image](https://github.com/patternfly/react-component-groups/assets/21317056/dfa93f53-4488-4e73-b556-08bbbb19f723)